cmd/kubectl-gadget/deploy.go

```diff
@@ -506,12 +546,26 @@ func runDeploy(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
+	// modify the otel-metrics-listen-address if specified
+	if otelMetricsListenAddr != "" {
+		// iterate over the objects and find the DaemonSet
+		// modify the prometheus.io/port annotation & break
+		for objectNum, obj := range objects {
+			if isDaemonSet(obj) {
```

Comment:
You can skip checking for DaemonSet if you move this check here: https://github.com/inspektor-gadget/inspektor-gadget/blob/main/cmd/kubectl-gadget/deploy.go#L614 ? I see you are already handling some annotations!